File: GigConnectBackend/src/main/java/com/gigconnect/service/PaymentService.java
package com.gigconnect.service;

import java.util.List;

import org.json.JSONObject;

import com.gigconnect.dtos.payment.CreateOrderRequestDto;
import com.gigconnect.dtos.payment.CreateOrderResponseDto;
import com.gigconnect.dtos.payment.PaymentFailedRequestDto;
import com.gigconnect.dtos.payment.PaymentResponseDto;
import com.gigconnect.dtos.payment.VerifyPaymentRequestDto;
import com.gigconnect.dtos.payment.VerifyPaymentResponseDto;

public interface PaymentService {
	CreateOrderResponseDto createOrder(CreateOrderRequestDto request) throws Exception;
	
	VerifyPaymentResponseDto verifyPayment(VerifyPaymentRequestDto request) throws Exception;

	void markPaymentFailed(PaymentFailedRequestDto request);
	
	List<PaymentResponseDto> getPaymentsByClient(Long clientId);
}

File: frontend/src/services/paymentService.js
import * as paymentApi from "../api/paymentApi";

export const createOrder = async (projectId) => {
    const response = await paymentApi.createOrder(projectId);
    return response.data;
};

export const verifyPayment = async (paymentData) => {
    const response = await paymentApi.verifyPayment(paymentData);
    return response.data;
};

export const markPaymentFailed = async (paymentData) => {
    const response = await paymentApi.markPaymentFailed(paymentData);
    return response.data;
};

export const getPaymentsByClient = async (clientId) => {
    const response = await paymentApi.getPaymentsByClient(clientId);
    return response.data;
};